pkg/l2announcer/l2announcer_test.go

```diff
@@ -338,19 +335,16 @@ func TestHappyPathPermutations(t *testing.T) {
 				assert.NoError(tt, err)
 			}
 
-			r = fix.proxyNeighborTable.Reader(fix.stateDB.ReadTxn())
-			iter, err = r.Get(statedb.All)
-			assert.NoError(tt, err)
-			entries = statedb.Collect[*tables.L2AnnounceEntry](iter)
+			rtx = fix.stateDB.ReadTxn()
+			iter, _ = fix.proxyNeighborTable.All(rtx)
+			entries = statedb2.Collect[*tables.L2AnnounceEntry](iter)
 			if assert.Len(tt, entries, 1) {
-				// We don't care about the amount of revisions.
-				entries[0].Revision = 1
 				assert.Equal(tt, entries[0], &tables.L2AnnounceEntry{
-					IP:               net.ParseIP(blueService().Spec.ExternalIPs[0]),
-					NetworkInterface: bluePolicy().Spec.Interfaces[0],
-					Origins:          []resource.Key{serviceKey(blueService())},
-					Deleted:          false,
-					Revision:         1,
+					L2AnnounceKey: tables.L2AnnounceKey{
+						IP:               netip.MustParseAddr(blueService().Spec.ExternalIPs[0]),
+						NetworkInterface: bluePolicy().Spec.Interfaces[0],
+					},
+					Origins: []resource.Key{serviceKey(blueService())},
```

Comment:
We may want to have a bit of a discussion around this style of testing if changing the structure of data requires changing tests. I realize it's somewhat in the vein of "testing the translation from one table to another", but I think it's usually a weakness in the tests if changing the target code requires reworking the various structure declarations in the test code.

Reply:
I agree with the shortcomings of this method, but not really sure how I would change things. Do you suggest some sort of constructor or is there some other technique? Looking forward to that discussion.